test/runtests.py

```diff
@@ -395,6 +395,59 @@ def on_delete(queue):
                                self.guardian.del_queue_size]
         self.assertTrue(len(queues_to_delete) == 0)
 
+    def test_delete_skip_unbounded(self):
+        self._setup_queue()
+
+        # Queue should still exist.
+        self._wait_for_queue()
+
+        # Get the queue's object
+        db_session.refresh(self.pulse_user)
+
+        self.assertTrue(len(self.pulse_user.queues) > 0)
+
+        # set queues as unbound so they won't be deleted
+        for queue in self.pulse_user.queues:
+            queue.unbound = 1
+
+        # Queue multiple messages while no consumer exists.
+        for i in xrange(self.guardian.del_queue_size + 1):
+            msg = self._build_message(i)
+            self.publisher.publish(msg)
+
+        # Wait some time for published messages to be taken into account.
+        for i in xrange(10):
+            time.sleep(0.3)
+            queues_to_delete = [q_data['name'] for q_data
+                                in pulse_management.queues()
+                                if q_data['messages_ready']
+                                   > self.guardian.del_queue_size]
+            if queues_to_delete:
+                break
+
+        # Test that there are some queues that should be deleted.
+        self.assertTrue(len(queues_to_delete) > 0)
+
+        # Setting up a callback to capture deleted queues
+        deleted_queues = []
+        def on_delete(queue):
+            deleted_queues.append(queue)
+        self.guardian.on_delete = on_delete
+
+        # Monitor the queues; this should create the queue object and assign
+        # it to the user.
+        for i in xrange(20):
+            self.guardian.monitor_queues(pulse_management.queues(),
+                                         pulse_management.bindings())
+            time.sleep(0.2)
```

Comment:
Again I realize this is the case in other tests, but we can add a check that exits this loop when everything's ready? I hate timing-based tests that assume things work (which I realize raises the question of why I approved the other tests... :)